includes/woopay/class-woopay-session.php

```diff
@@ -535,7 +524,13 @@ private static function has_valid_request_signature() {
 	 * @return bool True if WooPay is enabled, false otherwise.
 	 */
 	private static function is_woopay_enabled(): bool {
-		return WC_Payments_Features::is_woopay_eligible() && 'yes' === WC_Payments::get_gateway()->get_option( 'platform_checkout', 'no' );
+		$payment_settings = get_option( 'woocommerce_woocommerce_payments_settings', [] );
+		$woopay_enabled   = 'no';
+		if ( is_array( $payment_settings ) && ! empty( $payment_settings['platform_checkout'] ) ) {
+			$woopay_enabled = $payment_settings['platform_checkout'];
+		}
```

Comment:
Minor nitpick; I think we can just use the null coalescing operator `??` here since [it doesn't throw errors on variables that don't exist](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce):

```sh
$ php --interactive
Interactive shell

php > echo $this_does_not_exist['foo']['bar'] ?? 'baz';
baz
php > $text = 'foo';
php > echo $text['foo']['bar'] ?? 'baz';
baz
php > $array = [ 'foo' => [] ];
php > echo $array['foo']['bar'] ?? 'baz';
baz
php > $array = [ 'foo' => [ 'bar' => 'foobar' ] ];
php > echo $array['foo']['bar'] ?? 'baz';
foobar
```

```suggestion
		$woopay_enabled   = get_option( 'woocommerce_woocommerce_payments_settings', [] )['platform_checkout'] ?? 'no';
```